pkgs/development/python-modules/gradio/default.nix

```diff
@@ -0,0 +1,192 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, writeText
+
+# pyproject
+, hatchling
+, hatch-requirements-txt
+, hatch-fancy-pypi-readme
+
+# runtime
+, aiohttp
+, altair
+, fastapi
+, ffmpy
+, markdown-it-py
+, mdit-py-plugins
+, markupsafe
+#, linkify-it-py
+#, mdit-py-plugins
+, matplotlib
+, numpy
+, orjson
+, pandas
+#, paramiko
+, pillow
+, pycryptodome
+, python-multipart
+, pydub
+, pyyaml
+, requests
+, uvicorn
+, jinja2
+, fsspec
+, httpx
+, pydantic
+, websockets
+, typing-extensions
+
+# check
+, pytestCheckHook
+, pytest-asyncio
+, mlflow
+, huggingface-hub
+, transformers
+, wandb
+, respx
+, scikitimage
+, shap
+, ipython
+, ffmpeg
+, vega_datasets
+, boto3
+}:
+
+buildPythonPackage rec {
+  pname = "gradio";
+  version = "3.20.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  # We use the Pypi release, as it provides prebuild webui assets,
+  # and its releases are also more frequent than github tags
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-oG97GwehyBWjWXzDqyfj+x2mAfM6OQhYKdA3j0Rv8Vs=";
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
+    altair
+    fastapi
+    ffmpy
+    markdown-it-py
+    mdit-py-plugins
+    markupsafe
+    matplotlib
+    numpy
+    orjson
+    pandas
+    #paramiko
+    pillow
+    pycryptodome
+    python-multipart
+    pydub
+    pyyaml
+    requests
+    uvicorn
+    jinja2
+    fsspec
+    httpx
+    pydantic
+    websockets
+    typing-extensions
+  ] ++ markdown-it-py.optional-dependencies.linkify;
+
+  nativeBuildInputs = [
+    hatchling
+    hatch-requirements-txt
+    hatch-fancy-pypi-readme
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+    pytest-asyncio
+    mlflow
+    #cometml # FIXME: enable once comet-ml is packaged
+    huggingface-hub
+    transformers
+    wandb
+    respx
+    scikitimage
+    shap
+    ipython
+    ffmpeg
+    vega_datasets
+    boto3
+   ];
+
+  # Add pytest hook skipping tests that access network, marking them as "Expected fail" (xfail)
+  # We additionally xfail FileNotFoundError, as gradio often fail to upload test files to pypi.
```

Comment:
Why not fetch from github than instead?

Reply:
The pypi build includes prebuildt web assets

Reply:
Could we fetch them separately? Python is very brittle and we really would like to run tests